src/main/java/com/liferay/faces/util/product/ProductFactory.java

```diff
@@ -15,61 +15,88 @@
  */
 package com.liferay.faces.util.product;
 
-import java.util.Iterator;
-import java.util.ServiceLoader;
+import javax.faces.FacesWrapper;
+import javax.faces.context.ExternalContext;
+import javax.faces.context.FacesContext;
+
+import com.liferay.faces.util.factory.FactoryExtensionFinder;
+import com.liferay.faces.util.product.internal.ProductFactoryImpl;
 
 
 /**
  * @author  Kyle Stiemann
  */
-public abstract class ProductFactory {
+public abstract class ProductFactory implements FacesWrapper<ProductFactory> {
```

Comment:
@stiemannkj1 I thought it wasn't possible to decorate ProductFactory since it had to be consulted prior to scanning for faces-config.xml files?